app/frontend/src/App.tsx

```diff
@@ -33,12 +38,40 @@ function App() {
         },
         onReceivedExtensionMiddleTierToolResponse: message => {
             const result: ToolResult = JSON.parse(message.tool_result);
-
             const files: GroundingFile[] = result.sources.map(x => {
                 return { id: x.chunk_id, name: x.title, content: x.chunk };
             });
 
-            setGroundingFiles(prev => [...prev, ...files]);
+            setGroundingFiles(prev => [...prev, ...files]); // Keep track of all files used in the session
+            setAssistantGroundingFiles(files); // Store the grounding files for the assistant
+        },
+        onReceivedInputAudioTranscriptionCompleted: message => {
+            // Update history with input audio transcription when completed
+            const newHistoryItem: HistoryItem = {
+                id: message.event_id,
+                transcript: message.transcript,
+                groundingFiles: [], // Assuming no grounding files are associated with the transcription completed
+                sender: "user", // Indicate that this message is from the user
+                timestamp: new Date() // Add timestamp
+            };
+            setHistory(prev => [...prev, newHistoryItem]);
+        },
+        onReceivedResponseDone: message => {
+            const transcript = message.response.output.map(output => output.content?.map(content => content.transcript).join(" ")).join(" ");
+            if (!transcript) {
+                return; // Skip adding the history item if the transcript is null or empty
```

Comment:
nit/suggestion: remove comment, don't seem to be adding value/clarity to the code